SI/Sem.04/Pract.04/Task12.cpp

```diff
@@ -1,35 +1,54 @@
-#include <iostream>
+// Task12.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
 
+#include <iostream>
+#include <cmath>
 using namespace std;
 
-int checkInput() {
-	int input;
-	do {
-		cout << "Enter input: " << endl;
-		cin >> input;
-	} while (input < 100 || input > 1000000);
+int smallestNumber(int a)
+{
+    for (int i = a; i >= 0; --i)
+    {
+
+        if ((int)(log(i) / log(2)) == log(i) / log(2))
```

Comment:
The task should be done without using log. Also, you can optimize your solution.